backend/controller/controller.go

```diff
@@ -594,6 +590,10 @@ nextModule:
 }
 
 func (s *Service) Call(ctx context.Context, req *connect.Request[ftlv1.CallRequest]) (*connect.Response[ftlv1.CallResponse], error) {
+	return s.CallWithRequest(ctx, req, optional.None[model.RequestKey](), "")
+}
+
+func (s *Service) CallWithRequest(ctx context.Context, req *connect.Request[ftlv1.CallRequest], key optional.Option[model.RequestKey], requestSource string) (*connect.Response[ftlv1.CallResponse], error) {
```

Comment:
At first I tried this using headers, but requestKey needs to be passed around while requestSource does not. Making it more explicit that you are passing these parameters in rather than propagating made sense to me.
Happy to change this though.

Reply:
I see what you're trying to do, but other than this method the Service type is entirely RPC implementation methods for the gRPC service. This should remain the case, though I don't have an alternative solution.

Reply:
Actually, just make this private and the problem is solved.

Reply:
that makes sense